pkg/maps/policymap/policymap.go

```diff
@@ -70,23 +70,21 @@ func (pe *PolicyEntry) String() string {
 // +k8s:deepcopy-gen=true
 // +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapKey
 type PolicyKey struct {
-	Identity         uint32
-	DestPort         uint16 // In network byte-order
-	Nexthdr          uint8
-	TrafficDirection uint8
+	Identity         uint32 `align:"sec_label"`
+	DestPort         uint16 `align:"dport"` // In network byte-order
+	Nexthdr          uint8  `align:"protocol"`
+	TrafficDirection uint8  `align:"egress"`
 }
 
 // PolicyEntry represents an entry in the BPF policy map for an endpoint. It must
 // match the layout of policy_entry in bpf/lib/common.h.
 // +k8s:deepcopy-gen=true
 // +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapValue
 type PolicyEntry struct {
-	ProxyPort uint16 // In network byte-order
-	Pad0      uint16
-	Pad1      uint16
-	Pad2      uint16
-	Packets   uint64
-	Bytes     uint64
+	ProxyPort uint16    `align:"proxy_port"` // In network byte-order
+	Pad       [3]uint16 `align:"pad"`
```

Comment:
I don't have a very good reason why, but in the past we've avoided golang arrays like this, particularly in the middle of a structure as we've hit some weird issues: https://github.com/cilium/cilium/issues/3491

Reply:
This is interesting. The internal representation of a Go array is just its elements sequentially placed in memory.

Do you suggest to keep the previous version? If yes, then we could unroll the padding in the `policy_entry` struct so that it could be validated by alignchecker.

Reply:
Ah, sorry I thought the comment below with @aanm resolved this.

If it were me, I'd keep the three pad fields and just change the `policy_entry` to match / satisfy alignchecker.

Reply:
Done.